system/Config/BaseService.php

```diff
@@ -131,10 +156,17 @@ public static function locator(bool $getShared = true)
 	{
 		if ($getShared)
 		{
-			return static::getSharedInstance('locator');
+			if (empty(static::$instances['locator']))
+			{
+				static::$instances['locator'] = new \CodeIgniter\Autoloader\FileLocator(
+					static::autoloader()
+				);
+			}
+
+			return static::$instances['locator'];
 		}
 
-		return new \CodeIgniter\Autoloader\FileLocator(new \Config\Autoload());
```

Comment:
To solve issue 1552, it all started in this line. Using a `new \Config\Autoload()` it was not possible to dynamically use Autoloader namespaces.